pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -188,14 +188,24 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return handler.handleErr(err)
 	}
 	// Gather status of components
-	var refComps []v1alpha1.TypedReference
+	var refComps []common.ComponentRef
 	for _, comp := range comps {
-		refComps = append(refComps, v1alpha1.TypedReference{
+		refComp := common.ComponentRef{
 			APIVersion: comp.APIVersion,
 			Kind:       comp.Kind,
 			Name:       comp.Name,
-			UID:        app.UID,
-		})
+			UID:        comp.UID,
+		}
+		// Add workload kind to component status. Related issue:
+		// https://github.com/oam-dev/kubevela/issues/1351
+		var workloadMap map[string]interface{}
+		workloadMap, err = oamutil.RawExtension2Map(&comp.Spec.Workload)
+		if err == nil {
+			if wlRef, ok := workloadMap["definitionRef"]; ok {
+				refComp.WorkloadDefinition = oamutil.Object2RawExtension(wlRef)
```

Comment:
why is it a raw Extention? Shouldn't it just be a string?

Reply:
It depends on what information is needed here.

I thought before that provider an object of `workloadDefinition` here could be better than just provide a workload kind, but now I find that  if we provide object of workload, it seems workload name repeates twice.

Maybe you are right, I will change to provide just a name of workload name rather than total object.